shared/src/androidMain/kotlin/dev/johnoreilly/confetti/work/SessionNotificationWorker.kt

```diff
@@ -4,43 +4,68 @@ import android.content.Context
 import androidx.work.Constraints
 import androidx.work.CoroutineWorker
 import androidx.work.ExistingPeriodicWorkPolicy
-import androidx.work.PeriodicWorkRequest
+import androidx.work.ExistingWorkPolicy
+import androidx.work.OneTimeWorkRequestBuilder
 import androidx.work.PeriodicWorkRequestBuilder
 import androidx.work.WorkManager
 import androidx.work.WorkerParameters
 import dev.johnoreilly.confetti.work.SessionNotificationSender.Companion.INTERVAL
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
 import kotlin.time.toJavaDuration
 
+// Using a custom constructor has caused a crash inside Koin.
+// For now, as a workaround, we use KoinComponent with an inject for getting the notifier instance.
 class SessionNotificationWorker(
     context: Context,
     workerParams: WorkerParameters,
-    private val notifier: SessionNotificationSender,
-) : CoroutineWorker(context, workerParams) {
+) : CoroutineWorker(context, workerParams), KoinComponent {
```

Comment:
Not sure why, but Koin is failing silently when using a custom constructor for this worker. For now, as a workaround, I'm using `KoinComponent`. Later we can investigate further.